### DISCUSSION TITLE
Does the demo app use tkey splitting?

### DISCUSSION THREAD

**Discussion by MichaelStandenASM** [Web3Auth/Web3Auth#600] [Help]
I'm running through your documentation about tkey splitting and also your demo app here: https://demo-app.web3auth.io/
Does this demo use key splitting? If so where are the shared stored in this demo? 
It feels like I would only have 1 share (via oauth). 

I've been comparing it to the process here https://tech.tor.us/#auth-sec

**Comment by rinzler99** [comment]
1.  The demo app uses Web3Auth sdk for constructing the splitted key in the browser.
2.  Web3Auth sdk uses openlogin and tkey modules for splitting and constructing the key in browser. (Also you can setup you own  openlogin and tkey using self hosting)
3. So, the demo app use key splitting as it is using web3auth sdk which by default uses openlogin and tkey to construct/split the keys.
4. Before setting up the MFA In demo app there are 2/2 shares : social login and your device. So you will need these two shares whenever logging into the demo app again.
5. Finally, where are the shares stored in this demo? : As metioned first share is your social login and second share is your device from where you logged in. These two shares are custodied to yourself only.

**Reply by edwardchew97** [reply]
> 5. Finally, where are the shares stored in this demo? : As metioned first share is your social login and second share is your device from where you logged in. These two shares are custodied to yourself only.

Hi there, thanks for the answer, I was trying to look for the device share but I couldn't find anything in the localStorage and it seems to me the localStorage tkey is an empty string but I can see there's full private key in localStorage, could you point me to where the tkey share is being stored?
